scripts/pr.py

```diff
@@ -0,0 +1,273 @@
+
+
+def requireRemote():
+    remote = raw_input("Please, specify the repository for PR's in format ':owner/:repo': \n")
+    return remote.lower()
+
+
+def requireFeatureName():
+    return raw_input("Type name of the feature you will work on: ").replace(" ", "-")
+
+
+def mapDict(f,d):
+    return dict(map(f, d.iteritems()))
+
+
+def validateValues(items):
+    (k,v) = items
+    if v is None:
+        print("Cannot infer the {0} value".format(k))
+        newV = raw_input("Please specify a value for the : {0}".format(k))
+        return (k,newV)
+    else:  
+        return (k,v)
+
+    
+def readFile(name):
+    f = open(name, 'a+')
+    content = f.read()
+    f.close()
+    return content
+
+        
+def execStdout(cmd):
+    try:
+        return subprocess.check_output(cmd)
+    except subprocess.CalledProcessError:
+        None
+
+
+def mapOptional(x,f):
+    if not x is None:
+        return f(x)
+    else:
+        x
+
+
+def gitName():
+    name = execStdout(["git", "config", "user.name"])
+    mapOptional(name, lambda x: x.stript())
+
+
+def gitCurrentBranch():
+    return execStdout(["git", "rev-parse", "--abbrev-ref", "HEAD"]).strip()
+
+
+def gitOrigin():
+    origin =  execStdout(["git", "config", "remote.origin.url"])
+    mapOptional(origin, lambda x: x.strip())
+
+
+def parseJson(content):
+    try:
+        return json.loads(content)
+    except ValueError:
+        return {}
+
+    
+def optionalGet(dict,k):
+    if dict is None:
+        None
+    else:
+        dict.get(k, None)
+
+
+def dictMerge(l,r):
+    res = l.copy()
+    res.update(r.iteritems())
+    return res
+
+
+def writeJson(dict, name):
+    f = open(name, 'w')
+    content = json.dumps(dict)
+    f.write(content)
+    f.close()
+
+    
+def execCmd(args):
+    return subprocess.call(args)
+                       
+
+    
+class Config:
+    def __init__(self):
+        globalConfig = self.loadConfig()
+        
+        if globalConfig.get(scriptPath, None) is None:
+            newConfig = self.createConfig()
+            self.globalConfig = dictMerge(globalConfig, newConfig)
+            writeJson(self.globalConfig, configPath)
+        else:
+            self.globalConfig = globalConfig
+
+        self.setupConfig()
+
+        
+    def setupConfig(self):
+        self.config = self.globalConfig[scriptPath]
+        for k,v in self.config.items():
+            setattr(self, k, v)
+        
+
+    def loadConfig(self):
+        try:
+            content = readFile(configPath)
+            globalConfig = parseJson(content)
+            return globalConfig
+        except TypeError:
+            {}
+
+
+    def createConfig(self):
+        data = {'githubName': gitName()}
+        checkedDict = mapDict(validateValues, data)
+        checkedDict['remote'] = requireRemote()
+        globalConfig = { scriptPath: checkedDict }
+        
+        return globalConfig
+
+    
+    
+class Tasks:
+    def __init__(self, config):
+        self.config = config
```

Comment:
At the moment you only use `self.config.name` and `self.config.remote` in this class. Pass those parameters explicitly and have a usable default.